libs/testserver/server.go

```diff
@@ -119,11 +119,23 @@ func (s *Server) Handle(pattern string, handler HandlerFunc) {
 				}
 			}
 
+			var reqBody any
+			if json.Valid(body) {
+				// If the request body is a valid JSON, typecast it to json.RawMessage.
+				// This way json.Marshal will ignore the body and serialize it
+				// as is, which is what we want because the body is already a JSON.
+				reqBody = json.RawMessage(body)
```

Comment:
I don't see where this condition holds in this PR.

Reply:
It was relevant for the empty body `""`. It's been separately patched in https://github.com/databricks/cli/pull/2333

Reply:
Also for the `"/oidc/v1/token"` which has a non JSON body: `grant_type=client_credentials\u0026scope=all-apis`